presto-native-execution/presto_cpp/main/TaskManager.cpp

```diff
@@ -325,21 +353,7 @@ std::unique_ptr<TaskInfo> TaskManager::createOrUpdateTask(
     infoRequest = prestoTask->infoRequest;
   }
 
-  for (const auto& entry : resultRequests) {
-    auto resultRequest = entry.second.get();
-
-    VLOG(1) << "Processing pending result request for task "
-            << resultRequest->taskId << ", buffer " << resultRequest->bufferId
-            << ", sequence " << resultRequest->token;
-
-    getData(
-        resultRequest->promise.lock(),
-        resultRequest->taskId,
-        resultRequest->bufferId,
-        resultRequest->token,
-        resultRequest->maxSize,
-        *bufferManager_);
-  }
+  processResultRequests(resultRequests);
 
   if (outputBuffers.type == protocol::BufferType::BROADCAST) {
     execTask->updateBroadcastOutputBuffers(
```

Comment:
Will this call suffer from the same problem if buffers have been removed already?

Reply:
@mbasmanova this could hit the same issue as well.

Reply:
Let's file an issue to follow up and fix this.